test/integration/deploy_service/deploy_service_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package deploy_service
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeployService(t *testing.T) {
+	bpt := tft.NewTFBlueprintTest(t)
+
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		// bpt.DefaultVerify(assert)
+
+		projectId := bpt.GetStringOutput("project_id")
+		location := bpt.GetStringOutput("location")
+		clusterName := bpt.GetStringOutput("cluster_name")
+		gcloud.Runf(t, "container clusters get-credentials %s --region %s --project %s", clusterName, location, projectId)
+		k8sOpts := k8s.KubectlOptions{}
+		listServices, err := k8s.RunKubectlAndGetOutputE(t, &k8sOpts, "get", "svc", "terraform-example", "-o", "json")
+		if err != nil {
+			t.Fatalf("Unable to run kubectl to get the number of Services. Err: %s", err)
+		}
+		kubeService := utils.ParseJSONResult(t, listServices)
+		serviceIp := kubeService.Get("status.loadBalancer.ingress").Array()[0].Get("ip")
+		serviceUrl := fmt.Sprintf("http://%s:8080", serviceIp)
+		response, err := http.Get(serviceUrl)
```

Comment:
Since the previous test checks for service to start around ten times, let's wrap this in a poll.
https://github.com/terraform-google-modules/terraform-google-bootstrap/blob/b4d5bbd5752e983208278cb00ba46d434a536730/test/integration/tf_cloudbuild_workspace_simple/tf_cloudbuild_workspace_simple_test.go#L112

Reply:
added the poll for retry